### ISSUE TITLE
If a plugin calls the get_mx hook with credentials, those credentials are logged in clear text

### ISSUE THREAD

**Issue by tabascoterrier** [haraka/Haraka#3144]
**Describe the bug**
If a plugin calls the `get_mx` hook with credentials (e.g. `auth_user` and `auth_pass`), those credentials are logged in clear text.

**Expected behavior**
Having sensitive information in log files isn't ideal, it would be better if the credentials were redacted or simply not logged.

**Observed behavior**
Example (json) Haraka log entry:
```json
{
	"level": "INFO",
	"uuid": "8F1B9991-9A22-4A4B-B2D3-13C578FB0EF3.1.1",
	"origin": "outbound",
	"message": "",
	"hook": "get_mx",
	"plugin": "my_plugin",
	"function": "forwarder_lookup",
	"params": "example.com",
	"retval": "OK",
	"msg": {
		"priority": 0,
		"exchange": "192.168.1.1",
		"port": "465",
		"auth_user": "user1",
		"auth_pass": "hunter2"
	}
}
```

**Steps To Reproduce**
Register a plugin such as the below, and send a mail
```js
exports.register = function () {
    plugin.register_hook('get_mx', 'forwarder_lookup');
}

exports.forwarder_lookup = async function(next, hmail, domain) {

    // https://haraka.github.io/core/Outbound#the-get_mx-hook
    next_hop_mx = {
        priority: 0, 
        exchange: '192.168.1.1',
        port: 465,
        auth_user: 'user1',
        auth_pass: 'hunter2'

    }

    return next(OK, next_hop_mx);
}
```

**System Info:**
Haraka | Haraka.js — Version: 2.8.28
 --- | :---
Node | v16.19.0
OS | Linux collector-test 5.15.0-57-generic #63-Ubuntu SMP Thu Nov 24 13:48:31 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
openssl | OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)

**Additional context**
This appears to be due to `log_run_item` in `plugins.js` - it dutifully logs the return of a plugin every time, and in this case it can involve sensitive information.

I'm working around this by just patching my local `log_run_item` with `if (hook === 'get_mx') return;`, as I don't need these messages logged, but that's probably not a solution that will suit everyone.

I'm not massively familiar with Haraka's internals (and I'm quite new to nodejs to be honest), but could plugins define an override for the log level that `log_run_item` uses?

Some brief testing here suggests that adding something like this to the user plugin:

```js
exports.log_level = 'logdebug';
```

and then in Haraka's `plugins.js` `log_run_item`, switching out 

```js
if (is_not_cont) log = 'loginfo';
```

to

```js
// Default to loginfo, but override with the a plugin's exported log_level if set
if (is_not_cont) log = ((item[0].log_level) ? item[0].log_level : 'loginfo');
```

seems to work as expected, and lets a plugin override the loglevel. This means I can set that plugin to a level that'll never log credentials in production.

I'm happy to do a PR if the approach sounds reasonable. All insight and advice gratefully received.

**Comment by msimerson** [comment]
I wonder if it wouldn't be just as good to modify log_run_item and skip logging anything that matches `/auth|pass/`?

**Comment by tabascoterrier** [comment]
> I wonder if it wouldn't be just as good to modify log_run_item and skip logging anything that matches `/auth|pass/`?

That would also be fine by me @msimerson 👍

**Comment by msimerson** [comment]
Maybe?

```diff
diff --git a/plugins.js b/plugins.js
index da0504c6..ad34bd85 100644
--- a/plugins.js
+++ b/plugins.js
@@ -572,11 +572,20 @@ function log_run_item (item, hook, retval, object, params, msg) {
             'function'  :  item[1],
             'params'    :  ((params) ? ((typeof params === 'string') ? params : params[0]) : ''),
             'retval'    : constants.translate(retval),
-            'msg'       :  ((msg) ? msg : ''),
+            'msg'       :  sanitized(msg),
         });
     }
 }
 
+function sanitize (msg) {
+    if (!msg) return ''
+    const sanitized = { ...msg }; // copy the message
+    for (const priv of ['password','auth_pass']) {
+        delete sanitized[priv]
+    }
+    return sanitized
+}
+
 function is_deny_retval (val) {
     switch (val) {
         case constants.deny:
```

**Comment by msimerson** [comment]
merged in #3328
